From: Markus Koschany <apo@debian.org>
Date: Sat, 21 Jan 2017 05:37:25 +0100
Subject: use system sqlite

---
 src/Makefile | 11 ++---------
 1 file changed, 2 insertions(+), 9 deletions(-)

diff --git a/src/Makefile b/src/Makefile
index 8df1a12..f51b7df 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -83,7 +83,7 @@ BIN_SUFFIX=_native
 endif
 endif
 CLIENT_INCLUDES= $(INCLUDES) -I/usr/X11R6/include `sdl2-config --cflags`
-CLIENT_LIBS= -Lenet -lenet -L/usr/X11R6/lib -lX11 `sdl2-config --libs` -lSDL2_image -lSDL2_mixer -lz -lGL
+CLIENT_LIBS= -Lenet -lenet -L/usr/X11R6/lib -lX11 `sdl2-config --libs` -lSDL2_image -lSDL2_mixer -lz -lGL -lsqlite3
 endif
 ifneq (,$(findstring linux,$(PLATFORM)))
 CLIENT_LIBS+= -lrt
@@ -99,7 +99,6 @@ CLIENT_OBJS= \
 	shared/stream.o \
 	shared/tools.o \
 	shared/zip.o \
-	support/sqlite3.o \
 	engine/bih.o \
 	engine/blend.o \
 	engine/blob.o \
@@ -161,7 +160,7 @@ SERVER_INCLUDES= -DSTANDALONE $(INCLUDES) -Iinclude
 SERVER_LIBS= -mwindows $(STD_LIBS) -L$(WINDLL) -L$(WINLIB) -lzlib1 -lenet -lws2_32 -lwinmm
 else
 SERVER_INCLUDES= -DSTANDALONE $(INCLUDES)
-SERVER_LIBS= -Lenet -lenet -lz
+SERVER_LIBS= -Lenet -lenet -lz -lsqlite3
 endif
 SERVER_OBJS= \
 	shared/crypto-standalone.o \
@@ -169,7 +168,6 @@ SERVER_OBJS= \
 	shared/stream-standalone.o \
 	shared/tools-standalone.o \
 	shared/zip-standalone.o \
-	support/sqlite3.o \
 	engine/command-standalone.o \
 	engine/irc-standalone.o \
 	engine/master-standalone.o \
@@ -205,9 +203,6 @@ clean: clean-enet clean-client clean-server clean-genkey
 %-standalone.o: %.cpp
 	$(CXX) $(CXXFLAGS) -c -o $@ $(subst -standalone.o,.cpp,$@)
 
-support/sqlite3.o: support/sqlite3.c
-	$(CC) -DSQLITE_THREADSAFE=0 -DSQLITE_OMIT_LOAD_EXTENSION -c -o support/sqlite3.o support/sqlite3.c
-
 $(CLIENT_OBJS): CXXFLAGS += $(CLIENT_INCLUDES)
 $(filter shared/%,$(CLIENT_OBJS)): $(filter shared/%,$(CLIENT_PCH))
 $(filter engine/%,$(CLIENT_OBJS)): $(filter engine/%,$(CLIENT_PCH))
@@ -389,7 +384,6 @@ engine/master.o: shared/tools.h shared/command.h shared/geom.h shared/ents.h
 engine/master.o: shared/glexts.h shared/glemu.h shared/iengine.h
 engine/master.o: shared/igame.h engine/irc.h engine/sound.h engine/world.h
 engine/master.o: engine/octa.h engine/lightmap.h engine/bih.h
-engine/master.o: engine/texture.h engine/model.h support/sqlite3.h
 engine/material.o: engine/engine.h engine/version.h shared/cube.h
 engine/material.o: shared/tools.h shared/command.h shared/geom.h
 engine/material.o: shared/ents.h shared/glexts.h shared/glemu.h
@@ -693,7 +687,6 @@ engine/irc-standalone.o: engine/irc.h engine/sound.h
 engine/master-standalone.o: engine/engine.h engine/version.h shared/cube.h
 engine/master-standalone.o: shared/tools.h shared/command.h shared/geom.h
 engine/master-standalone.o: shared/ents.h shared/iengine.h shared/igame.h
-engine/master-standalone.o: engine/irc.h engine/sound.h support/sqlite3.h
 engine/server-standalone.o: engine/engine.h engine/version.h shared/cube.h
 engine/server-standalone.o: shared/tools.h shared/command.h shared/geom.h
 engine/server-standalone.o: shared/ents.h shared/iengine.h shared/igame.h
